File: mainActivity/src/main/java/net/airvantage/model/UserRights.java
package net.airvantage.model;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

import android.content.Context;

import com.sierrawireless.avphone.R;

public class UserRights extends ArrayList<String> {

    private static final long serialVersionUID = 1L;

    private static Map<String, Integer> AS_STRING_ID = new HashMap<>();

    static {
        AS_STRING_ID.put("entities.systems.view", R.string.entities_systems_view);
        AS_STRING_ID.put("entities.systems.create", R.string.entities_systems_create);
        AS_STRING_ID.put("entities.systems.edit", R.string.entities_systems_edit);
        AS_STRING_ID.put("entities.applications.view", R.string.entities_applications_view);
        AS_STRING_ID.put("entities.applications.create", R.string.entities_applications_create);
        AS_STRING_ID.put("entities.applications.edit", R.string.entities_applications_edit);
        AS_STRING_ID.put("entities.alerts.rule.view", R.string.entities_alerts_rule_view);
        AS_STRING_ID.put("entities.alerts.rule.create.edit.delete", R.string.entities_alerts_rule_create_edit_delete);
    }

    public static String asString(String rightKey, Context context) {
        String res = null;
        Integer stringId = AS_STRING_ID.get(rightKey);
        if (stringId != null) {
            res = context.getString(stringId);
        }
        return res;
    }

}

File: mainActivity/src/main/java/net/airvantage/model/UserRights.kt
package net.airvantage.model

import java.util.ArrayList
import java.util.HashMap

import android.content.Context

import com.sierrawireless.avphone.R

class UserRights : ArrayList<String>() {
    companion object {

        private const val serialVersionUID = 1L

        private val AS_STRING_ID = HashMap<String, Int>()

        init {
            AS_STRING_ID["entities.systems.view"] = R.string.entities_systems_view
            AS_STRING_ID["entities.systems.create"] = R.string.entities_systems_create
            AS_STRING_ID["entities.systems.edit"] = R.string.entities_systems_edit
            AS_STRING_ID["entities.applications.view"] = R.string.entities_applications_view
            AS_STRING_ID["entities.applications.create"] = R.string.entities_applications_create
            AS_STRING_ID["entities.applications.edit"] = R.string.entities_applications_edit
            AS_STRING_ID["entities.alerts.rule.view"] = R.string.entities_alerts_rule_view
            AS_STRING_ID["entities.alerts.rule.create.edit.delete"] = R.string.entities_alerts_rule_create_edit_delete
        }

        fun asString(rightKey: String, context: Context): String? {
            var res: String? = null
            val stringId = AS_STRING_ID[rightKey]
            if (stringId != null) {
                res = context.getString(stringId)
            }
            return res
        }
    }

}
